tests/modules/update.py

```diff
@@ -288,6 +292,67 @@ def test_update_different_branch_mix_modules_branch_test(self):
     assert modules_json.get_module_version("multiqc", GITLAB_URL, GITLAB_REPO) == GITLAB_BRANCH_TEST_NEW_SHA
 
 
+# Mock questionary answer: do not update module, only show diffs
+@mock.patch.object(questionary.Question, "unsafe_ask", return_value=False)
+def test_update_only_show_differences(self, mock_prompt):
+    """Try updating all modules showing differences.
+    Don't update some of them.
+    Check that the sha in modules.json is not changed."""
+    modules_json = ModulesJson(self.pipeline_dir)
+    update_obj = ModuleUpdate(self.pipeline_dir, update_all=True, show_diff=True)
+
+    assert update_obj.update() is True
+
+    mod_json = modules_json.get_modules_json()
+    # Loop through all modules and check that they are NOT updated (according to the modules.json file)
+    # Modules that can be updated but shouldn't are custom/dumpsoftwareversions and fastqc
+    # Module multiqc is already up to date so don't check
```

Comment:
Wasn't the problem that it touched files, even when we excluded them, so should we maybe also check that Multiqc didn't change?

Reply:
The problem was that it was updating the sha in `modules.json` but not updating the files. But we could also check for files unchanged to have more control